packages/tooling/src/lib/parameters-to-json-schema.js

```diff
@@ -123,107 +92,31 @@ function getOtherParams(pathOperation, oas) {
     operationParams = operationParams.concat(commonParamsNotInParams || []);
   }
 
-  const resolvedParameters = operationParams.map(param => {
-    if (param.$ref) return findSchemaDefinition(param.$ref, oas);
-    return param;
-  });
-
-  const constructSchema = data => {
-    const schema = {};
-
-    if (data.type === 'array') {
-      schema.type = 'array';
-
-      if ('items' in data) {
-        if (Object.keys(data.items).length === 1 && typeof data.items.$ref !== 'undefined') {
-          schema.items = findSchemaDefinition(data.items.$ref, oas);
-        } else {
-          schema.items = data.items;
-        }
-
-        // Run through the arrays contents and clean them up.
-        schema.items = constructSchema(schema.items);
-      } else if ('properties' in data || 'additionalProperties' in data) {
-        // This is a fix to handle cases where someone may have typod `items` as `properties` on an array. Since
-        // throwing a complete failure isn't ideal, we can see that they meant for the type to be `object`, so we can do
-        // our best to shape the data into what they were intendint it to be.
-        // README-6R
-        schema.type = 'object';
-      } else {
-        // This is a fix to handle cases where we have a malformed array with no `items` property present.
-        // README-8E
-        schema.items = {};
-      }
-    }
-
-    if (data.type === 'object') {
-      schema.type = 'object';
-
-      if ('properties' in data) {
-        schema.properties = {};
-
-        Object.keys(data.properties).map(prop => {
-          schema.properties[prop] = constructSchema(data.properties[prop]);
-          return true;
-        });
-      }
-
-      if ('additionalProperties' in data) {
-        if (typeof data.additionalProperties === 'object' && data.additionalProperties !== null) {
-          schema.additionalProperties = constructSchema(data.additionalProperties);
-        } else if (data.additionalProperties !== false) {
-          // If it's set to `false`, don't bother adding it.
-          schema.additionalProperties = data.additionalProperties;
-        }
-      }
-    }
-
-    if ('allowEmptyValue' in data) {
-      schema.allowEmptyValue = data.allowEmptyValue;
-    }
-
-    // Only add a default value if we actually have one.
-    if (typeof data.default !== 'undefined') {
-      if ('allowEmptyValue' in schema && schema.allowEmptyValue && data.default === '') {
-        // If we have `allowEmptyValue` present, and the default is actually an empty string, let it through as it's
-        // allowed.
-        schema.default = data.default;
-      } else if (data.default !== '') {
-        schema.default = data.default;
-      }
-    }
-
-    if ('maxLength' in data) schema.maximum = data.maxLength;
-    if ('minLength' in data) schema.minimum = data.minLength;
```

Comment:
This was actually a large misunderstanding on my part with `maxLength` and `minLength`. OAS supports those two properties, as well as `minimum` and `maximum`. `maxLength` and `minLength` are to be used on strings, whereas `maximum` and `minimum` are for numbers. Mixing them like I was previously doing here wrong.